applications/MappingApplication/custom_utilities/interface_condition.h

```diff
@@ -93,89 +89,206 @@ namespace Kratos
+                  && mNumPoints == 2) { // I am a linear line condition
+                  is_inside = MapperUtilities::ProjectPointToLine(mpCondition, rGlobalCoords,
                                                                   projection_local_coords,
-                                                                  projection_distance, m_normal);
-          } else if (m_geometry_family == GeometryData::Kratos_Quadrilateral) { // I am a quadrilateral condition
-              is_inside = MapperUtilities::ProjectPointToQuaddrilateral(m_p_condition, global_coords,
-                                                                        projection_local_coords,
-                                                                        projection_distance, m_normal);
-          } else {
-              KRATOS_ERROR << "MappingApplication; InterfaceCondition; \"EvaluateResult\" Unsupported Geometry!" << std::endl;
+                                                                  projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Triangle 
+                        && mNumPoints == 3) { // I am a linear triangular condition
+                  is_inside = MapperUtilities::ProjectPointToTriangle(mpCondition, rGlobalCoords,
+                                                                      projection_local_coords,
+                                                                      projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Quadrilateral
+                        && mNumPoints == 4) { // I am a linear quadrilateral condition
+                  is_inside = MapperUtilities::ProjectPointToQuadrilateral(mpCondition, rGlobalCoords,
+                                                                          projection_local_coords,
+                                                                          projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Tetrahedra ||
+                         mGeometryFamily == GeometryData::Kratos_Prism ||
+                         mGeometryFamily == GeometryData::Kratos_Hexahedra) { // Volume Mapping
+                  is_inside = MapperUtilities::PointLocalCoordinatesInVolume(mpCondition, rGlobalCoords,
+                                                                             projection_local_coords,
+                                                                             projection_distance);
+              } else {
+                  std::cout << "MAPPER WARNING, Unsupported geometry, " 
+                            << "using an approximation" << std::endl;
+                  return false;
+              }
+          } catch (...) {
+              std::cout << "MAPPER WARNING, Functions necessary for projection "
+                        << "are not implemented in the core, " 
+                        << "using an approximation" << std::endl;
+              return false;
           }
```

Comment:
I'm sorry but I don't understand the code then. As I'm reading it, you are checking the element type and, once the element is identified as known, you are calling the appropriate function in mapper utilities. Neither the if blocks that check the element type nor the functions in mapper utilities throw errors (and in particular, none of them throw errors related to missing implementations, as the error message suggests), so I don't see what you are trying to catch here.

In any case, I'd be wary of using try/catch for normal, expected behavior. Exceptions can have unexpected side effects (for example, exceptions finish execution of functions at unexpected points, which can prevent properly deallocating memory) and, in my opinion, should be reserved for exceptional circumstances, not normal operation.